packages/vuetify/src/components/VDivider/VDivider.ts

```diff
@@ -1,42 +1,40 @@
-// @ts-nocheck
-/* eslint-disable */
-
 // Styles
 import './VDivider.sass'
 
-// Types
-import { VNode } from 'vue'
+// Utilities
+import { defineComponent, h } from 'vue'
 
-// Mixins
-import Themeable from '../../mixins/themeable'
+// Composables
+import { useTheme } from '@/composables'
 
-export default Themeable.extend({
-  name: 'v-divider',
+export default defineComponent({
+  name: 'VDivider',
 
   props: {
     inset: Boolean,
     vertical: Boolean,
   },
 
-  render (h): VNode {
-    // WAI-ARIA attributes
-    let orientation
-    if (!this.$attrs.role || this.$attrs.role === 'separator') {
-      orientation = this.vertical ? 'vertical' : 'horizontal'
-    }
-    return h('hr', {
-      class: {
-        'v-divider': true,
-        'v-divider--inset': this.inset,
-        'v-divider--vertical': this.vertical,
-        ...this.themeClasses,
-      },
-      attrs: {
-        role: 'separator',
-        'aria-orientation': orientation,
-        ...this.$attrs,
-      },
-      on: this.$listeners,
-    })
+  setup (props, { attrs }) {
+    const { themeClasses } = useTheme()
+    const role = attrs.role ?? 'separator'
+    const ariaOrientation = role === 'separator'
+      ? props.vertical ? 'vertical' : 'horizontal'
```

Comment:
No idea why `attrs.role` seems to be reactive here, but `props.vertical` is not (`aria-orientation` doesn't change to `vertical`):

```vue
<template>
  <v-app>
    <div style="width: 100px; height: 100px">
      <v-divider :vertical="vertical" />
    </div>
    <button @click="vertical = !vertical">toggle vertical</button>
  </v-app>
</template>

<script>
  export default {
    data: () => ({ vertical: false }),
  }
</script>
```